doc/console/SONiC-Console-Switch-High-Level-Design.md

```diff
@@ -224,7 +266,59 @@ ssh tom@2001:db8::2
+
+For example: `ttyUSB0`
+
+Usually, tty devices will mapping to each console device port, but the sequence might not aligned with hardware designed port number which means when you notice that there is a ttyUSB0, it probably not binding to panel's port 0.
+
+Below is an example, we can look into the inner topology of a console device:
+
+![Underlying Port-Device Mapping](./UnderlyingPortDeviceMapping.png)
+
+The console device have 4 com ports at front panel with product port numbers from 1 to 4. After we connected the console device to SONiC switch, the SONiC OS will detect 4 tty device and put them under `/dev` directory.
+
+From SONiC aspect, we are hard to aware how the tty device mapped to physical console port because its depends on the external hardware design and OS kernel behavior. Then we need to create alias for all tty devices by using some rule. Here[[1]](https://github.com/Azure/SONiC/blob/master/doc/udev-terminalserver/udev%20rules%20for%20Terminal%20Server.md) is an design for resolving USB terminal server's port mapping problem by leveraging udev rule, it use the usb hardware metadata which are immutable to determine the relationship between underlying tty devices and front panel ports.
+
+Regardless of the console device type, the rule is required. We define below function `g()` for determining the alias name for a console port:
+
+```
+tty_device_alias_name <- g(console_port_name)
+```
+
+The rule will map each hardware port to an alias tty device which name were generated by above function `g()`. Once the rule been applied, the underlying hardware detail and origin tty devices will be transparent for all upper layer applications.
+
+In other hand, no matter how many add-on devices and how you connect to the console device. We treat them as a single box with SONiC switch which means you need to create aliasing rules for all your add-on devices. Once the mapping finalized, we can create alias name for all tty devices, we can use `g()` to find the alias tty device under `\dev` directory.
+
+Below is an example. Assume that we have three add-on console devices which are using different driver. Each device have 2 com ports at frontend.
+
+![Sample Multiple Console Devices](./SampeMultipleConsoleDevice.png)
+
+The console switch designer should create a description document which can explain how the physical console port mapped to SONiC console port.
+
+For example, we can create below mapping table, it defined the sequence of console port in SONiC switch's angle. Obviously, the order and port mapping should be determined by the SONiC console switch designer.
+
+#### Table 2: Underlying device mapping definition sample
+
+| tty device name | product port | console port name  | tty device alias |
+|---|---|---|---|---|
+| ttyUSB0 | 1 | 0 | ttySONiC0 |
+| ttyUSB1 | 2 | 1 | ttySONiC1 |
+| ttyACM0 | 1 | 2  | ttySONiC2 |
+| ttyACM1 | 2 | 3  | ttySONiC3 |
+| ttyUSB2 | 1 | 4  | ttySONiC4 |
+| ttyUSB3 | 2 | 5  | ttySONiC5 |
```

Comment:
Need to quote this table with ```

Reply:
This is a mark down table. I referenced it at the `[Table of Contents](#table-of-contents)` and suppose it should be quote with code block.